libs/medic-users-meta/index.js

```diff
@@ -1,15 +1,45 @@
 const Postgrator = require('postgrator');
 const log = require('loglevel');
 
-const migrate = postgresUrl => {
-  const postgrator = new Postgrator({
-    migrationDirectory: `${__dirname}/migrations`,
+const migrate = async (postgresUrl) => {
+
+  const postgratorConfig = {
     schemaTable: 'users_meta_migrations',
     driver: 'pg',
     logProgress: log.getLevel() <= log.levels.DEBUG,
     connectionString: postgresUrl,
+  };
+
+  const postgrator = new Postgrator({
+    ...postgratorConfig,
+    migrationDirectory: `${__dirname}/migrations`,
   });
 
+  let lastMigrationExecuted = '';
+  try {
+    lastMigrationExecuted = await postgrator.getDatabaseVersion();
+  } catch (e) {
+    if (e.message !== 'relation "users_meta_migrations" does not exist') {
+      throw e;  // Unknown error
+    }
+    // First time migrations for medic-users-meta are executed, so
+    // migrations under ./migrations/deprecated were not executed,
+    // therefore no need to execute ./migrations/patch scripts
+  }
+  if (lastMigrationExecuted === '202102191153') {
+    // The first migration that contains bugs was executed earlier,
+    // the migrations under ./migrations/patch are executed to drop
+    // the views without dropping the tables with the data
+    const postgratorPatch = new Postgrator({
```

Comment:
I'm not sure I follow. Maybe we don't have to drop the table. Just the materialized view and re-create it. If that's possible, I'd rather do that and keep the code clean without a bunch of hardcoded stuff relating to migration dates and so on.

Reply:
> I'm not sure I follow. Maybe we don't have to drop the table.

Yes, actually the table were not dropped, only the views (and its dependents indexes that were also part of the problem).

> I'd rather do that and keep the code clean without a bunch of hardcoded stuff relating to migration dates and so on.

Yes, my fault, I rolled back the changes and added just the changes needed in one script. For some reason testing different scenarios: - installing with clean DBs - installing with existent DB running last stable version - same but the previous stable version that don't have the meta views... and in some of the tests I got an error like the migration couldn't run because previous data already imported, but I guess I misinterpreted the results.

I tested again with the latest changes all the possible scenarios and it worked. Also after run the migrations for the first time, I tried stopping the the process, adding and removing records in Couch, and restarting the process did refresh not just the data but the materialized vires (I could wait also the process to re-run without stopping).